preload/vperm_filestatgates.c

```diff
@@ -354,7 +370,7 @@ int fchownat_gate(int *result_errno_ptr,
 				res = -1;
 			}
 		} else {
-			*result_errno_ptr = errno;
+			*result_errno_ptr = e;
```

Comment:
So this relies on the result being non-zero when the `result_errno_ptr` is used by the caller? Just thinking about the Clang analyzer complaints on ConnMan as these were reported as garbage returns, since the error variable is not always touched.

Reply:
This change seems to replicate a similar error found in a fork of a contributor ldbox:
https://github.com/ldbox/ldbox/commit/e8875c5a802424b5b23fe4304f6ca201b2bf6646

(Only commenting this here for reference not as critique or alike).